### PR TITLE
Add output shape on a block. Add squared shape.

### PR DESCRIPTION
**Pull request by samelhusseini**
## The basics
- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Fixes https://github.com/google/blockly/issues/3414

### Proposed Changes

Add ouput shape property on a block and parse from json.
Add implementation for a square shape in zelos. 
Use a block's output shape if one exists when deciding on what shape to use.

### Reason for Changes

Zelos rendering. 

### Test Coverage

Tested with pxt-blockly blocks in zelos rendering playground.

Tested on:

### CODE REVIEW

File: core/renderers/zelos/constants.js [hunk 1]
```diff
@@ -515,15 +516,77 @@ Blockly.zelos.ConstantProvider.prototype.makeRounded = function() {
   };
 };
 
+/**
+ * @return {!Object} An object containing sizing and path information about
```
Comment (alschmiedt) [inline core/renderers/zelos/constants.js]: Missing function description

File: core/renderers/zelos/constants.js [hunk 2]
```diff
@@ -515,15 +516,77 @@ Blockly.zelos.ConstantProvider.prototype.makeRounded = function() {
...
+ *     a squared shape for connections.
+ * @package
+ */
+Blockly.zelos.ConstantProvider.prototype.makeSquared = function() {
+  var radius = this.CORNER_RADIUS;
+
+  // The 'up' and 'down' versions of the paths are the same, but the Y sign
+  // flips.  Forward and back are the signs to use to move the cursor in the
+  // direction that the path is being drawn.
```
Comment (alschmiedt) [inline core/renderers/zelos/constants.js]: Could you rephrase this comment? I'm not entirely sure what this part means: Forward and back are the signs to use

**Comment by samelhusseini** [comment]
@alschmiedt I've fixed up the comments for how all the shapes are drawn (rounded, hexagonal and squared). Let me know if those comments make sense.

**Comment by mattbishop** [comment]
Also a picture would be nice...

**Comment by samelhusseini** [comment]
Sample of a square shaped block:
Screen Shot 2020-01-07 at 1 54 43 PM

File: core/renderers/zelos/constants.js [hunk 3]
```diff
@@ -517,16 +523,20 @@ Blockly.zelos.ConstantProvider.prototype.makeRounded = function() {
...
  */
 Blockly.zelos.ConstantProvider.prototype.makeSquared = function() {
   var radius = this.CORNER_RADIUS;
 
-  // The 'up' and 'down' versions of the paths are the same, but the Y sign
-  // flips.  Forward and back are the signs to use to move the cursor in the
-  // direction that the path is being drawn.
+  // The main path for the squared connection shape is made out of two corners
+  // and a single line in-between (a and v). These are defined in relative
+  // positions and require the hight of the block.
```
Comment (alschmiedt) [inline core/renderers/zelos/constants.js]: nit: *height

File: core/renderers/zelos/constants.js [hunk 4]
```diff
@@ -517,16 +523,20 @@ Blockly.zelos.ConstantProvider.prototype.makeRounded = function() {
...
+  // The 'left' and 'right' versions of the paths are the same, but the Y sign
+  // flips.  The 'up' and 'down' versions of the path determine where the corner
+  // point is placed and in-turn how the direction of the corners.
```
Comment (alschmiedt) [inline core/renderers/zelos/constants.js]: nit: I think this is suppose to be "and in-turn the direction of the corners"?

File: core/renderers/zelos/constants.js (hunk 3 above)
Comment (samelhusseini) [reply core/renderers/zelos/constants.js]: Done.
